corehq/ex-submodules/casexml/apps/phone/models.py

```diff
@@ -700,10 +700,11 @@ def get_cases_that_directly_depend_on_case(self, case_id):
         return self.reverse_indices.get(case_id, [])
 
     def delete_index(self, from_case_id, index_name):
-        prior_ids = self.indices.pop(from_case_id, {})
-        prior_ids.pop(index_name, None)
-        if prior_ids:
-            self.indices[from_case_id] = prior_ids
+        case_indices = self.indices.get(from_case_id, {})
+        if len(case_indices) > 1:
+            case_indices.pop(index_name, {})
+        elif len(case_indices) == 1:
+            self.indices.pop(from_case_id)
```

Comment:
aaand now that I write this comment, I notice that this is actually a reference to a different `models.py:708`

https://github.com/dimagi/commcare-hq/blob/master/corehq/form_processor/models.py#L708-L717